feature-libs/user/profile/components/update-profile/update-profile.component.ts

```diff
@@ -1,47 +1,40 @@
-import { Component } from '@angular/core';
-import {
-  GlobalMessageService,
-  GlobalMessageType,
-  RoutingService,
-} from '@spartacus/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
 import { User } from '@spartacus/user/account/root';
-import { Title, UserProfileFacade } from '@spartacus/user/profile/root';
-import { BehaviorSubject, Observable } from 'rxjs';
+import { Title } from '@spartacus/user/profile/root';
+import { Observable, Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { UpdateProfileService } from './update-profile.service';
 
 @Component({
   selector: 'cx-update-profile',
   templateUrl: './update-profile.component.html',
 })
-export class UpdateProfileComponent {
-  titles$: Observable<Title[]> = this.userProfile.getTitles();
-  user$: Observable<User | undefined> = this.userProfile.get();
-  isLoading$ = new BehaviorSubject(false);
+export class UpdateProfileComponent implements OnInit, OnDestroy {
+  constructor(protected service: UpdateProfileService) {}
 
-  constructor(
-    private routingService: RoutingService,
-    private userProfile: UserProfileFacade,
-    private globalMessageService: GlobalMessageService
-  ) {}
+  form: FormGroup = this.service.form;
+  isUpdating$ = this.service.isUpdating$;
+  titles$: Observable<Title[]> = this.service.titles$;
+  user$: Observable<User | undefined> = this.service.user$;
 
-  onSuccess(): void {
-    this.globalMessageService.add(
-      { key: 'updateProfileForm.profileUpdateSuccess' },
-      GlobalMessageType.MSG_TYPE_CONFIRMATION
-    );
-    this.routingService.go({ cxRoute: 'home' });
-  }
+  private subscription: Subscription;
 
-  onCancel(): void {
-    this.routingService.go({ cxRoute: 'home' });
+  ngOnInit() {
+    this.form.disable();
+    this.subscription = this.user$
```

Comment:
I'm wondering about moving this logic to service, and maybe even trying to drop the subscribction by tying it to the isUpdating$ stream? 🤔